### PR TITLE
testing/openjdk12: New aport

### PR DESCRIPTION
**Pull request by bratkartoffel**
This PR adds the latest OpenJDK (12) release.

### CODE REVIEW

**Comment by bratkartoffel** [comment]
@tcely why the "broken" label?

**Comment by tcely** [comment]
@bratkartoffel because of the multiple CI failures

**Comment by bratkartoffel** [comment]
@tcely please remove dependencies label, I've removed the 32 bit arches

**Comment by bratkartoffel** [comment]
Can someone please review & merge this PR?
It's based on my other PRs with openjdk9-11, mainly the patch offsets had to be adjusted.

File: testing/openjdk12/Alpine_Bug_10126.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+public class Alpine_Bug_10126 {
+    public static void main(String[] args) throws Exception {
+        try (java.net.Socket sock = javax.net.ssl.SSLSocketFactory.getDefault().createSocket("bugs.alpinelinux.org", 443);
```
Comment (tcely) [inline testing/openjdk12/Alpine_Bug_10126.java]: Connecting to bugs is not the best idea now.

File: testing/openjdk12/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,307 @@
+# Contributor: Simon Frankenberger <simon-alpine@fraho.eu>
+# Maintainer: Simon Frankenberger <simon-alpine@fraho.eu>
+pkgname=openjdk12
+pkgver=12.0.2_p10
+_pkgver=${pkgver/_p/+}
+pkgrel=0
+pkgdesc="Oracle OpenJDK 12"
+url="https://hg.openjdk.java.net/jdk-updates/jdk12u"
+arch="all !x86 !armhf !armv7" # openjdk11 is not yet available on 32 bit arches
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]: How does openjdk11 relate?

File: testing/openjdk12/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+libffi-dev
+libjpeg-turbo-dev
+libx11-dev
+libxext-dev
+libxrandr-dev
+libxrender-dev
+libxt-dev
+libxtst-dev
+linux-headers
+zlib-dev"
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]:
```suggestion
	zlib-dev
	"
```

Please indent lists.

File: testing/openjdk12/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+build.patch
+aarch64.patch
+arm.patch
+ppc64le.patch
+x86.patch
+
+HelloWorld.java
+TestECDSA.java
+TestCryptoLevel.java
+Alpine_Bug_10126.java
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]:
```suggestion
	Alpine_Bug_10126.java
```

Please indent lists.

File: testing/openjdk12/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+	fi
+
+	# CFLAGS, CXXFLAGS and LDFLAGS are ignored as shown by a warning
+	# in the output of ./configure unless used like such:
+	#  --with-extra-cflags="${CFLAGS}"
+	#  --with-extra-cxxflags="${CXXFLAGS}"
+	#  --with-extra-ldflags="${LDFLAGS}"
+	# See also paragraph "Configure Control Variables" from "common/doc/building.md"
+	# shellcheck disable=2097 disable=2098
+	CFLAGS='' CXXFLAGS='' LDFLAGS='' \
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]: Why is assigning empty values to these environment variables desirable?

File: testing/openjdk12/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+		--with-jvm-variants=server \
+		--with-debug-level=release \
+		--with-version-pre= \
+		--with-version-opt=alpine-r${pkgrel} \
+		--with-version-build=${_pkgver#*+} \
+		--with-vendor-name="Alpine" \
+		--with-vendor-url="https://alpinelinux.org/" \
+		--with-vendor-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues" \
+		--with-vendor-vm-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues"
+	MAKEFLAGS='' make jdk-image
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]: Why assign `MAKEFLAGS` here?

File: testing/openjdk12/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+		--enable-dtrace=no \
+		--with-jvm-variants=server \
+		--with-debug-level=release \
+		--with-version-pre= \
+		--with-version-opt=alpine-r${pkgrel} \
+		--with-version-build=${_pkgver#*+} \
+		--with-vendor-name="Alpine" \
+		--with-vendor-url="https://alpinelinux.org/" \
+		--with-vendor-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues" \
+		--with-vendor-vm-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues"
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]: Update bugs URLs?

File: testing/openjdk12/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+		_logfile=$( mktemp -p "$builddir" )
+		MAKEFLAGS='' xvfb-run make \
+			run-test-tier1 \
+			run-test-tier2 \
+			run-test-tier3 \
+			| tee "$_logfile"
+		msg "---------------------------------------"
+		msg "The build log can be found at $_logfile"
+		# abort the build so you may take a look at the logfile
+		false
```
Comment (tcely) [inline testing/openjdk12/APKBUILD]:
```suggestion
		return 1
```

File: testing/openjdk12/Alpine_Bug_10126.java (hunk 1 above)
Comment (bratkartoffel) [reply testing/openjdk12/Alpine_Bug_10126.java]: Thanks, updated

File: testing/openjdk12/APKBUILD (hunk 2 above)
Comment (bratkartoffel) [reply testing/openjdk12/APKBUILD]: The openjdk builds depend on the previous JDK. As OpenJDK11 is only available for 64 bit arches this blocks this aport

File: testing/openjdk12/APKBUILD (hunk 3 above)
Comment (bratkartoffel) [reply testing/openjdk12/APKBUILD]: done

File: testing/openjdk12/APKBUILD (hunk 5 above)
Comment (bratkartoffel) [reply testing/openjdk12/APKBUILD]: If these variables are set, the configuration prints ugly warnings. See the comment above

File: testing/openjdk12/APKBUILD (hunk 6 above)
Comment (bratkartoffel) [reply testing/openjdk12/APKBUILD]: When `MAKEFLAGS` are specified, the build fails as it cannot handle the `-j` option. So setting it to empty makes the build work

Comment (tcely) [reply testing/openjdk12/APKBUILD]: It sounds like adding `-j 1` may be a better solution.

**Comment by bratkartoffel** [comment]
Openjdk build does not support the makeflags and issues a warning if it's set.

Am 6. Oktober 2019 22:40:49 MESZ schrieb tcely <notifications@github.com>:
>tcely commented on this pull request.
>
>
>
>> +		--with-test-jobs=${JOBS:-4} \
>+		--with-native-debug-symbols=external \
>+		$_with_jtreg \
>+		--disable-warnings-as-errors \
>+		--disable-precompiled-headers \
>+		--enable-dtrace=no \
>+		--with-jvm-variants=server \
>+		--with-debug-level=release \
>+		--with-version-pre= \
>+		--with-version-opt=alpine-r${pkgrel} \
>+		--with-version-build=${_pkgver#*+} \
>+		--with-vendor-name="Alpine" \
>+		--with-vendor-url="https://alpinelinux.org/" \
>+		--with-vendor-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues"
>\
>+		--with-vendor-vm-bug-url="https://bugs.alpinelinux.org/projects/alpine/issues"
>+	MAKEFLAGS='' make jdk-image
>
>It sounds like adding `-j 1` may be a better solution.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/alpinelinux/aports/pull/8552#discussion_r331809268

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by bratkartoffel** [comment]
This PR is ready to merge now, I've dropped 32 bit support again as discussed in #11631

**Pull request closed without merging**